web/source/kmwosk.js

```diff
@@ -944,69 +1051,132 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
     osk.getModifierState = function(layerId)
     {
         var modifier=0;
-        if(layerId.indexOf('shift') >= 0) modifier += 1;
-        if(layerId.indexOf('ctrl')  >= 0) modifier += 2;
-        if(layerId.indexOf('alt')   >= 0) modifier += 4;
+        if(layerId.indexOf('shift') >= 0) {
+          modifier += osk.modifierCodes["SHIFT"];
```

Comment:
Let's use `|=` to clarify that these are bitmasks

Reply:
Can do.  Thought the original was a bit funny, but when in doubt I try to perturb the original style as little as possible.